presto-main/src/test/java/com/facebook/presto/operator/TestMergeOperator.java

```diff
@@ -0,0 +1,382 @@
+import com.facebook.presto.execution.buffer.TestingPagesSerdeFactory;
+import com.facebook.presto.metadata.RemoteTransactionHandle;
+import com.facebook.presto.metadata.Split;
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.block.SortOrder;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.split.RemoteSplit;
+import com.facebook.presto.sql.planner.plan.PlanNodeId;
+import com.facebook.presto.testing.MaterializedResult;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.ImmutableList;
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.testing.TestingHttpClient;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.facebook.presto.RowPagesBuilder.rowPagesBuilder;
+import static com.facebook.presto.SessionTestUtils.TEST_SESSION;
+import static com.facebook.presto.operator.OperatorAssertion.assertOperatorIsBlocked;
+import static com.facebook.presto.operator.OperatorAssertion.assertOperatorIsUnblocked;
+import static com.facebook.presto.operator.OperatorAssertion.toMaterializedResult;
+import static com.facebook.presto.operator.PageAssertions.assertPageEquals;
+import static com.facebook.presto.spi.block.SortOrder.ASC_NULLS_FIRST;
+import static com.facebook.presto.spi.block.SortOrder.DESC_NULLS_FIRST;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.testing.TestingTaskContext.createTaskContext;
+import static com.facebook.presto.testing.assertions.Assert.assertEquals;
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static java.util.concurrent.Executors.newScheduledThreadPool;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+@Test(singleThreaded = true)
+public class TestMergeOperator
+{
+    private static final String TASK_1_ID = "task1";
+    private static final String TASK_2_ID = "task2";
+    private static final String TASK_3_ID = "task3";
+
+    private ScheduledExecutorService executor;
+    private PagesSerdeFactory serdeFactory;
+    private HttpClient httpClient;
+    private ExchangeClientFactory exchangeClientFactory;
+    private AtomicInteger operatorId = new AtomicInteger();
+
+    private LoadingCache<String, TestingTaskBuffer> taskBuffers;
+
+    @BeforeMethod
+    public void setUp()
+    {
+        executor = newScheduledThreadPool(10, daemonThreadsNamed("test-merge-operator-%s"));
```

Comment:
nit: do we really need 10 threads in the pool (given that we have memory issues in travis)?